handler/asset_allocations_handler.go

```diff
@@ -1,13 +1,127 @@
 package handler
 
 import (
+	"encoding/json"
+	"fmt"
 	"net/http"
 
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/vkhichar/asset-management/contract"
+	"github.com/vkhichar/asset-management/customerrors"
 	"github.com/vkhichar/asset-management/service"
 )
 
-func CreateAssetAllocationHandler(assetAllocation service.AssetAllocationService) http.HandlerFunc {
+func CreateAssetAllocationHandler(assetAllocationService service.AssetAllocationService) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+
+		assetId, err := uuid.Parse(vars["asset_id"])
+
+		if err != nil {
+			fmt.Printf("handler:incorrect asset id")
+			w.WriteHeader(http.StatusBadRequest)
+			responseBytes, _ := json.Marshal(contract.ErrorResponse{Error: "invalid asset id"})
+			w.Write(responseBytes)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+
+		var request contract.CreateAssetAllocationRequestForJson
```

Comment:
we can get user_id from request params like change /assets/{asset_id}/allocate to /assets/{asset_id}/allocate/{user_id} , this makes more sense